dynamic_config/dynamic_config_manager.py

```diff
@@ -0,0 +1,195 @@
+"""Component that handles dynamic configuration of the Envoy proxy."""
+
+import json
+import yaml
+from enum import Enum
+from google.protobuf import json_format
+
+from abc import ABC, abstractmethod
+import envoy.config.cluster.v3.cluster_pb2 as cluster_pb2
+import envoy.service.discovery.v3.discovery_pb2 as discovery_pb2
+import envoy.extensions.upstreams.http.v3.http_protocol_options_pb2 as http_protocol_options_pb2
+import envoy.extensions.transport_sockets.tls.v3.tls_pb2 as tls_pb2
+import nighthawk.api.configuration.cluster_config_manager_pb2 as cluster_manager_pb2
+
+import random
+
+from google.protobuf import json_format
+import argparse
+import time
+import base64
+import logging
+import os
+
+
+class DynamicConfigManager(ABC):
+  """Base class for Dynamic configuration components."""
```

Comment:
If we do have a good reason to keep this as an interface, can we expand the commentary here outlining what the interface is for and what rules (if any) should the implementations follow?